routes.go

```diff
@@ -68,18 +68,17 @@ func provideCoreOption(server string, in RoutesIn) arrangehttp.Option[http.Serve
 
 			mux := chi.NewMux()
 
-			// TODO: should probably customize things a bit
-			mux.Use(recovery.Middleware(recovery.WithStatusCode(555)))
-
 			options := []otelmux.Option{
 				otelmux.WithTracerProvider(in.Tracing.TracerProvider()),
 				otelmux.WithPropagators(in.Tracing.Propagator()),
 			}
 
+			// TODO: should probably customize things a bit
+			mux.Use(recovery.Middleware(recovery.WithStatusCode(555)), otelmux.Middleware("server_primary", options...),
+				candlelight.EchoFirstTraceNodeInfo(in.Tracing.Propagator(), true))
```

Comment:
this will cause an annoying bug
```suggestion
				candlelight.EchoFirstTraceNodeInfo(in.Tracing.Propagator(), false))
```